version.go

```diff
@@ -19,6 +19,16 @@ const VersionRegexpRaw string = `v?([0-9]+(\.[0-9]+)*?)` +
 	`(\+([0-9A-Za-z\-]+(\.[0-9A-Za-z\-]+)*))?` +
 	`?`
 
+type VersionPart int
+
+const (
+	MajorPart VersionPart = iota
```

Comment:
Borrow from this Pull Request:-
https://github.com/hashicorp/go-version/pull/9/files#diff-ccde124315e74e68fae484840d08dff5R25